src/olympia/files/models.py

```diff
@@ -431,8 +431,15 @@ def webext_permissions_list(self):
             return []
         try:
             # Filter out any errant non-strings included in the manifest JSON.
-            return [p for p in self._webext_permissions.permissions
-                    if isinstance(p, basestring)]
+            # Remove any duplicate permissions.
+            permissions = [p for p in self._webext_permissions.permissions
+                           if isinstance(p, basestring)]
+            filtered_permissions = set()
+            filtered_permissions_add = filtered_permissions.add
```

Comment:
I think resolving filtered_permissions.add in each iteration is more costly than resolving a local variable.Am I right?